### PR TITLE
feat: build package in macOS

### PR DESCRIPTION
**Pull request by kmilo9999**
- Added Macos on the test workflows

### CODE REVIEW

File: .github/workflows/python_ci.yml [hunk 1]
```diff
@@ -6,10 +6,15 @@ jobs:
...
+    runs-on:  ${{ matrix.os }}
     steps:
-      - name: install mpi
-        run: sudo apt update && sudo apt-get install openmpi-bin openmpi-doc libopenmpi-dev 
+      - name: install mpi in ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: sudo apt update && sudo apt-get install openmpi-bin openmpi-doc libopenmpi-dev
+      - name: install mpi macos
+        if: matrix.os == 'macos-latest'
+        run: brew install open-mpi
```
Comment (cpaniaguam) [inline .github/workflows/python_ci.yml:17]: Clarifying question: Does this directive install all deps that the other Linux command does?

Comment (kmilo9999) [reply .github/workflows/python_ci.yml:17]: It should. 
Well, now that I take a look at the linux command, dependencies `openmpi-doc` and `libopenmpi-dev` are extra.

Comment (cpaniaguam) [reply .github/workflows/python_ci.yml:17]: 'Extra' meaning 'not needed'?

**Review by cpaniaguam — APPROVED** [review]
Great!

File: .github/workflows/python_ci.yml (hunk 1 above)
Comment (kmilo9999) [reply .github/workflows/python_ci.yml:17]: Correct!

**Review by hollandjg — APPROVED** [review]
Great!

**Pull request merged**